src/main/java/org/mule/service/scheduler/internal/threads/DedicatedSchedulerThreadPools.java

```diff
@@ -108,6 +108,8 @@ protected void doStart(boolean preStartThreads) throws MuleException {
                                new SchedulerThreadFactory(computationGroup),
                                byCallerThreadGroupPolicy.apply(computationGroup.getName()));
 
+    preserveThreadOnErrorGroups = new HashSet<>(asList(ioGroup, customWaitGroup, customCallerRunsAnsWaitGroup));
```

Comment:
in `createThreadGroupPolicy` this same set is already created. Extract and reuse.

Reply:
Done.